cranelift-reader/src/lexer.rs

```diff
@@ -399,6 +401,27 @@ impl<'a> Lexer<'a> {
         token(Token::Name(&self.source[begin..end]), loc)
     }
 
+    /// Scan for a multi-line quoted string with no escape character.
+    fn scan_string(&mut self) -> Result<LocatedToken<'a>, LocatedError> {
+        let loc = self.loc();
+        let begin = self.pos + 1;
+
+        assert_eq!(self.lookahead, Some('"'));
+
+        while let Some(c) = self.next_ch() {
+            if c == '"' || c == '\n' {
+                break;
+            }
+        }
+
+        let end = self.pos;
+        if self.lookahead != Some('"') {
```

Comment:
If we exited the previous loop because c = `\n`, this will yield an error, I think, which is undesired since this says this can handle multi-line strings. If so, we should remove the `|| c == '\n'` part of the if check in the while above. Can you add a test scanning a string containing a new line, please?

Reply:
Done.